packages/angular/src/rule-builder/rule.component.ts

```diff
@@ -0,0 +1,127 @@
+import { Component, EventEmitter, HostBinding, Input, Output } from '@angular/core';
+import { ListItem } from 'carbon-components-angular';
+
+@Component({
+  selector: 'ai-rule',
+  template: `
+    <ng-container *ngIf="isRuleGroup">
+      <ai-rule-builder-group-logic [id]="rule.id" [(selected)]="rule.groupLogic">
+      </ai-rule-builder-group-logic>
+      <ng-container *ngFor="let r of rule.rules; let i = index">
+        <ai-rule
+          (addRule)="addRule.emit($event)"
+          (removeRule)="removeRule.emit($event)"
+          [columns]="columns"
+          [columnOperands]="columnOperands"
+          [(rule)]="rule.rules[i]"
+        ></ai-rule>
+      </ng-container>
+    </ng-container>
+    <ng-container *ngIf="isRule">
+      <ibm-dropdown
+        theme="light"
+        placeholder="Select a column"
+        [(ngModel)]="rule.columnId"
+        value="id"
+      >
+        <ibm-dropdown-list [items]="getColumns()"></ibm-dropdown-list>
+      </ibm-dropdown>
+      <ibm-dropdown
+        theme="light"
+        placeholder="Select an operand"
+        [(ngModel)]="rule.operand"
+        value="id"
+      >
+        <ibm-dropdown-list [items]="getColumnOperands()"></ibm-dropdown-list>
+      </ibm-dropdown>
+      <input
+        *ngIf="!hasTemplate()"
+        ibmText
+        theme="light"
+        placeholder="Enter a value"
+        [(ngModel)]="rule.value"
+      />
+      <ng-template
+        *ngIf="hasTemplate()"
+        [ngTemplateOutlet]="getTemplate()"
+        [ngTemplateOutletContext]="{ $implicit: rule }"
+      >
+      </ng-template>
+      <div class="iot--rule-builder-rule__actions">
+        <button ibmButton="ghost" [iconOnly]="true" (click)="removeRule.emit(rule.id)">
+          <svg class="bx--btn__icon" ibmIcon="subtract" size="32"></svg>
+          <span class="bx--assistive-text">Remove rule</span>
```

Comment:
Is it too much to have this and the other assistive texts in this template be modifiable as well?

Reply:
Well, doh.